src/endpoints/ECatCourse/index.ts

```diff
@@ -744,6 +756,384 @@ class ECatCourse extends EndpointCategory {
+              .map((migrationIssue: any, i: number) => {
+                if (i === migrationIssues.length - 1) {
+                  return `and ${migrationIssue.description}`;
+                }
+                return migrationIssue.description;
+              })
+            // Put together
+              .join(', ')
+          );
+        }
+
+        const errorMessage = `We ran into an error while migrating your course content: ${errorsAsText}.`;
+
+        throw new CACCLError({
+          message: errorMessage,
+          code: ErrorCode.MigrationIssue,
+        });
+      }
+    } catch (err) {
+      if (err instanceof CACCLError) {
+        // Rethrow the error (it's already in the right format)
+        throw err;
+      }
+      // An unknown error occurred. Throw a new error
+      throw new CACCLError({
+        message: err,
+        code: ErrorCode.MigrationIssue,
+      });
+    }
+    let sourceAssignments = await this.api.course.assignment.list({
+      courseId: sourceCourseId,
+    });
+    // filter sourceAssignments to only those that were migrated
+    sourceAssignments = sourceAssignments.filter((assignment) => { return assignmentIds.includes(assignment.id); });
+
+    const destinationAssignments = await this.api.course.assignment.list({
+      courseId: destinationCourseId,
+    });
+
+    // mapping source group id to destination group id
+    const assignmentGroupMap: { [k: number]: number } = {};
+
+    // mapping source assignment id to destination assignment id
+    const assignmentMap: { [k: number]: number } = {};
+    // iterate through each source assignment to determine the mapping
+    sourceAssignments.forEach((sourceAssignment) => {
+      const destinationAssignment = destinationAssignments.find((assignment) => {
+        return assignment.name === sourceAssignment.name;
+      });
+      if (destinationAssignment) {
+        assignmentMap[sourceAssignment.id] = destinationAssignment.id;
+      } else {
+        throw new CACCLError({
+          message: 'Could not find a migrated assignment in the destination course.',
+          code: ErrorCode.MigrationIssue,
+        });
+      }
+    });
+
+    // iterate through each assignment group in the source course and
+    // create the same assignment group in the destination course
+    const sourceAssignmentGroups = await this.api.course.assignmentGroup.list({
+      courseId: sourceCourseId,
+    });
+    const sourceAssignmentGroupIds = sourceAssignmentGroups.map((group) => { return group.id; });
+    for (let i = 0; i < sourceAssignmentGroupIds.length; i++) {
+      const sourceId = sourceAssignmentGroupIds[i];
+      const sourceAssignmentGroup = await this.api.course.assignmentGroup.get({
+        assignmentGroupId: sourceId,
+        courseId: sourceCourseId,
+      });
+      // check if apply_assignment_group_weights is true in the source course
+      const sourceCourse = await this.visitEndpoint({
```

Comment:
Use the built-in function: `api.course.get`

Plus, you probably don't need to do this inside the loop

Reply:
`this.api.course.get({ courseId: sourceCourseId })`